src/components/autocompleter/autocompleter.jsx

```diff
@@ -0,0 +1,106 @@
+import PropTypes from 'prop-types';
+import React, { forwardRef, useEffect, useState } from 'react';
+import useFetch from '@bloodyaugust/use-fetch';
+import Select from '../select/select.jsx';
+import ListOption from '../list-option/list-option.jsx';
+import mockConstants from '../../mocks/mock-constants.js';
+
+const { API_ROOT } = mockConstants;
+
+const Autocompleter = forwardRef(function Autocompleter(props, ref) {
+  const { execute: fetch } = useFetch();
+  const [models, setModels] = useState(props.models);
+
+  useEffect(() => {
+    fetchModels();
+  }, []);
+
+  function fetchModels(searchString) {
+    fetch(apiEndpoint(searchString))
+      .then((results) => {
+        setModels(results.json.results.map(resultInfo => results.json[resultInfo.key][resultInfo.id]));
+      }).catch((error) => {
+        if (error.error && error.error.type !== 'aborted') {
+          throw error;
+        }
+      });
+  }
+
+  function apiEndpoint(searchString) {
+    if (searchString) {
+      return `${API_ROOT}${props.apiEndpoint}?${props.searchParam}=${searchString}`;
+    }
+
+    return `${API_ROOT}${props.apiEndpoint}`;
+  }
+
+  function displayName(modelInfo) {
+    return modelInfo.title || modelInfo.name || modelInfo.full_name || modelInfo.currency;
```

Comment:
`currency` is jarring to me here, one thing we could provide is a way for the consumer of this to supply a prop that is something like a key to know what to present, ie 

```

<Metacompleter labelKey="currency" />

function displayName(modelInfo) { 
  if(props.labelKey) { return modelInfo[labelKey]; }

  return modelInfo.title || modelInfo.name || modelInfo.full_name;
}
```

That way we don't need to try to guess at what it could possibly be. In a lot of cases, `title` and `name` will cover it, and `user` is used so often that `full_name` is nice to have here, but there will always be edge cases where someone decides to name the "user-friendly" name of something as like `display_name`, `capitalized_name`, or a one-off like `currency` is.

Reply:
This is an excellent call. Let's take it one step further and make it so the autocompleter-user can pass in a function or a key. And then let's have this nasty logic `modelInfo.title || modelInfo.name || modelInfo.full_name` live in our feature in bigmaven.

Then, we / someone doesn't have to make an mds pr and do the whole song and dance when we want to improve / add to this list.